src/WeatherApp.jsx

```diff
@@ -1,50 +1,308 @@
-
-    return () => {
-      setCity("");
-      setWeatherData({});
-    };
+            setCityValue(name);
+            setWeatherData((prevWeatherData) => ({
+              ...prevWeatherData,
+              ...createDataObject(data),
+            }));
+          })
+          .finally(() => {
+            setIsLoading(false);
+          });
+      });
+    }
+    return () => {};
   }, []);
 
+  useEffect(() => {
+    // Keeps the current weather icon updated whenever a change is induced.
+    updateWeatherIcon();
+  }, [weatherData]);
+
   const handleInputChange = (event) => {
+    const isDropDown = event.target.tagName === "SELECT";
+    setCityValue(event.target.value);
     setCity(event.target.value);
+    if (isDropDown) {
+      getCityInfo(event.target.value).then(([lat, lon]) => {
+        getWeatherData(lat, lon, tempVal)
+          .then((data) => {
+            setWeatherData((prevWeatherData) => ({
+              ...prevWeatherData,
+              ...createDataObject(data, tempVal),
+            }));
+          })
+          .finally(() => {
+            setIsLoading(false);
+          });
+      });
+    }
+  };
+  const updateWeatherIcon = () => {
+    const newIcon = weatherIcon();
+    setIcon(newIcon);
+  };
+
+  const createDataObject = (data, unit = "metric") => {
+    try {
+      const { list } = data;
+      const [firstDataSet] = list;
+      const { temp, weather, humidity: Humidity } = firstDataSet;
+      const {
+        day: Day,
+        min: Min,
+        max: Max,
+        night: Night,
+        eve: Evening,
+        morn: Morning,
+      } = temp;
+      const temperatures = {
+        Morning,
+        Day,
+        Evening,
+        Night,
+        Min,
+        Max,
+      };
+      const [weatherObject] = weather;
+      const { main: CurrentWeather } = weatherObject;
+
+      const tempSign = unit === "metric" ? "°" : "°F";
+
+      for (const key in temperatures) {
+        temperatures[key] = temperatures[key] + tempSign;
+      }
+
+      const dataObject = {
+        CurrentWeather,
+        ...temperatures,
+        Humidity,
+      };
+
+      return dataObject;
+    } catch (error) {
+      throw new Error("Error Creating Data Object");
+    }
+  };
+
+  const tempChange = () => {
+    const newTemp = tempVal === "metric" ? "imperial" : "metric";
+
+    setTemp(newTemp);
+    convertValues(newTemp);
+  };
+
+  const convertValues = (unit) => {
+    getCityInfo(cityNameValue).then(([lat, lon]) => {
+      getWeatherData(lat, lon, unit)
+        .then((data) => {
+          setWeatherData((prevWeatherData) => ({
+            ...prevWeatherData,
+            ...createDataObject(data, unit),
+          }));
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    });
+  };
+
+  const weatherIcon = () => {
+    // TO render the current weather description icon
+    const iconStyle = { fontSize: "10vw" };
+
+    let icon;
+
+    switch (weatherData.CurrentWeather) {
```

Comment:
Create WeatherIcon into a separate component,
create a Map for the WatherNames
return one component "FontAwesomeIcon"